### PR TITLE
Support meta scan in BE

### PR DESCRIPTION
**Pull request by laotan332**
1. support get meta info like min or max, which got by read data page before. now it can be read from segment footer
2. collect dict info if the column data is low cardinality

### CODE REVIEW

**Comment by kangkaisen** [comment]
For #559

File: be/src/runtime/vectorized/statistic_result_writer.cpp [hunk 1]
```diff
@@ -84,12 +88,39 @@ Status StatisticResultWriter::append_chunk(vectorized::Chunk* chunk) {
...
+    }
+}
+
 void StatisticResultWriter::_fill_statistic_data_v1(int version, const vectorized::Columns& columns,
                                                     const vectorized::Chunk* chunk, TFetchDataResult* result) {
     SCOPED_TIMER(_serialize_timer);
 
     // mapping with Data.thrift.TStatisticData
-    DCHECK(columns.size() == 11);
+    DCHECK(columns.size() == 11 || columns.size() == 12);
```
Comment (stdpain) [inline be/src/runtime/vectorized/statistic_result_writer.cpp:123]: remove change in this function

File: be/src/runtime/vectorized/statistic_result_writer.cpp [hunk 2]
```diff
@@ -84,12 +88,39 @@ Status StatisticResultWriter::append_chunk(vectorized::Chunk* chunk) {
...
+    auto versioncolumn = ColumnHelper::cast_to_raw<TYPE_BIGINT>(columns[1]);
+    auto dictColumnViewer = ColumnViewer<TYPE_VARCHAR>(columns[2]);
+
+    std::vector<TStatisticData> data_list;
+    int num_rows = chunk->num_rows();
+    data_list.resize(num_rows);
+
+    for (int i = 0; i < num_rows; ++i) {
+        data_list[i].__set_meta_version(versioncolumn->get_data()[i]);
+        if (!dictColumnViewer.is_null(i)) {
```
Comment (stdpain) [inline be/src/runtime/vectorized/statistic_result_writer.cpp:104]: implement deserilize TGlobalDict reference 
be/test/exprs/agg/aggregate_test.cpp: test_dict_merge

**Review by stdpain — DISMISSED** [review]

File: be/src/storage/vectorized/meta_read.cpp [hunk 3]
```diff
@@ -0,0 +1,353 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "storage/vectorized/meta_read.h"
```
Comment (dirtysalt) [inline be/src/storage/vectorized/meta_read.cpp:3]: Is 'meta_reader.h' more appropriate?

File: be/src/runtime/vectorized/statistic_result_writer.cpp [hunk 4]
```diff
@@ -14,6 +15,7 @@
 namespace starrocks::vectorized {
 
 const int STATISTIC_DATA_VERSION1 = 1;
+const int STATISTIC_DATA_VERSION2 = 101;
```
Comment (dirtysalt) [inline be/src/runtime/vectorized/statistic_result_writer.cpp:18]: Why so large gap between v1 and v2?

File: be/src/runtime/vectorized/statistic_result_writer.cpp [hunk 5]
```diff
@@ -84,6 +88,34 @@ Status StatisticResultWriter::append_chunk(vectorized::Chunk* chunk) {
     return status;
 }
 
+void StatisticResultWriter::_fill_statistic_data_v2(int version, const vectorized::Columns& columns,
```
Comment (dirtysalt) [inline be/src/runtime/vectorized/statistic_result_writer.cpp:91]: Where the statistics data will be stored into ?

Comment (dirtysalt) [reply be/src/runtime/vectorized/statistic_result_writer.cpp:91]: Is that just viewable or be used somewhere?

**Review by dirtysalt — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

File: be/src/runtime/vectorized/statistic_result_writer.h [hunk 6]
```diff
@@ -35,6 +35,8 @@ class StatisticResultWriter final : public ResultWriter {
 
     void _fill_statistic_data_v1(int version, const vectorized::Columns& columns, const vectorized::Chunk* chunk,
                                  TFetchDataResult* result);
+    void _fill_statistic_data_v2(int version, const vectorized::Columns& columns, const vectorized::Chunk* chunk,
```
Comment (kangkaisen) [inline be/src/runtime/vectorized/statistic_result_writer.h:38]:
```suggestion
    void _fill_dict_statistic_data(int version, const vectorized::Columns& columns, const vectorized::Chunk* chunk,
```

File: be/src/storage/vectorized/meta_reader.h [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    void check_validation() const {
+        if (UNLIKELY(version.first == -1)) {
+            LOG(FATAL) << "version is not set. tablet=" << tablet->full_name();
+        }
+    }
+
+    const std::map<int32_t, std::string>* id_to_names = nullptr;
+    const DescriptorTbl* desc_tbl = nullptr;
+
+    int chunk_size = 1024;
```
Comment (kangkaisen) [inline be/src/storage/vectorized/meta_reader.h:41]: config:vector_chunk_size?

**Review by kangkaisen — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**